bson/_cbsonmodule.c

```diff
@@ -1747,6 +1747,9 @@ int write_dict(PyObject* self, buffer_t buffer,
                     PyObject *etype = NULL, *evalue = NULL, *etrace = NULL;
                     PyErr_Fetch(&etype, &evalue, &etrace);
                     PyObject *InvalidDocument = _error("InvalidDocument");
+                    if (InvalidDocument == NULL) {
```

Comment:
I just realized there's a larger problem here. This logic is only in place for the `is_dict` fast path where it should be added generically for any mapping.

Reply:
Okay, I'll see if I can factor this out.